WifiART/src/userinterface/UsageView.java

```diff
@@ -67,4 +68,73 @@ private void setup() {
         this.add(staLocationPolicyPanelPanel);
     }
 
+    // Getters
+
+    /**
+     * Get the value of the RetransmissionAllowed parameter from the view.
+     * @return the value of the RetransmissionAllowed parameter.
+     */
+    public boolean getRetransmissionAllowed() {
+        return retransmissionAllowedCheckbox.isSelected();
+    }
+
+    /**
+     * Get the value of the RetentionExpires parameter from the view.
+     * @return the value of the RetentionExpires parameter.
+     */
+    public boolean getRetentionExpires() {
+        return retentionExpiresCheckbox.isSelected();
+    }
+
+    /**
+     * Get the value of the ExpireTime parameter from the view.
+     * @return the value of the ExpireTime parameter.
+     * @throws NumberFormatException If the user did not enter an integer value.
+     */
+    public int getExpireTime() throws NumberFormatException {
+        return Integer.parseInt(expireTimeField.getText());
+    }
+
+    /**
+     * Get the value of the StaLocationPolicy parameter from the view.
+     * @return the value of the StaLocationPolicy parameter.
+     */
+    public boolean getStaLocationPolicy() {
+        return staLocationPolicyCheckbox.isSelected();
+    }
+
+    // Methods for adding listeners
+
+    /**
+     * Add a listener for the RetransmissionAllowed parameter.
+     * @param listener the ActionListener for the RetransmissionAllowed parameter.
+     */
+    public void setRetransmissionAllowedListener(ActionListener listener) {
```

Comment:
Should this be addRetransmissionAllowedListener instead? 

- setXXXListener usually indicates only one listener is allowed, and settings a new listeners would replace the old listener. 
- addXXXListener means multiple listeners are allowed, and a new listener doesn't interfere with previously added listeners.

Reply:
I agree that "add" is better since you can have multiple listeners. In this specific case I don't think it makes much of a difference since there isn't a reason to have multiple listeners per field.

Reply:
Its more standard to use addXXXListener. Also you don't know what will be added in the future, so this makes it more easily extensible.

Reply:
This will be fixed in the newest revision.